tests/installation/bootloader_svirt.pm

```diff
@@ -345,7 +345,7 @@ sub run {
 
     # connects to a guest VNC session
     select_console('sut', await_console => 0);
-    vmware_set_permanent_boot_device($boot_device);
+    vmware_set_permanent_boot_device($boot_device) unless (check_var('UEFI', '1'));
```

Comment:
I suggest to add a new function in lib/Utils/Backend.pm:

sub is_uefi_boot {

    if (check_var('UEFI', '1') or check_var('IPXE_UEFI', '1')) {
        return 1;
    }
    return 0;
}

Then call it here.

Implement is_uefi_boot in lib/Utils/Backends.pm

Reply:
In my perspective, there are two reasons which does not require to check the variable 'IPXE_UEFI'.
1. I read the description in _lib/Utils/Backend.pm_ which is used for checking different backend or console types,  it's a bit strange to include an unrelated function.
2. My understanding, the variable 'IPXE_UEFI' is for baremetal host installation, but vmware tests are based on VM installation. From the variable name, it can't represent the setting of guest UEFI boot, and we do not support guest with ipxe boot at present.

Reply:
1.I think lib/utils/Backend.pm is still the right place. If you do not think so, you can also consider lib/Utils/Architectures.pm. 
2.It does not matter, because is_uefi_boot checks whether system is in UEFI mdoe, either parameter, UEFI or IPXE_UEFI can solely determine this as long as test uses them. So you implement this function and call it in your test really helps. It does not matter which parameter is being used. If you use of them, then it is in UEFI mode.

Reply:
is_uefi_boot by checking UEFI=1 or IPXE_UEFI=1 is not very acurate, in case of our prj1 gi_xxx test suites, if this function is used to check the guest, it would be not correct. I think checking UEFI=1 at here is more reasonable.

Reply:
> is_uefi_boot by checking UEFI=1 or IPXE_UEFI=1 is not very acurate, in case of our prj1 gi_xxx test suites, if this function is used to check the guest, it would be not correct. I think checking UEFI=1 at here is more reasonable.

If parameter checking directly with UEFI=1 is more reasonable, then using is_uefi_boot is better, because it provides better support and can be reused repeatedly without trouble. If you have concern about guest, then this can be further improved later, but it does not impact inventing is_uefi_boot.

Reply:
I can fully understand, but here the main point is only one parameter "UEFI" is enough for checking vmware guest boot option. In addition, one more parameter will cause the tests confused, e.g. Other tests like JeOS doesn't give the parameter "UEFI", but for some reason "IPXE_UEFI" is set to 1. If it's really necessary checking more parameters, then we can create a separate function for instead.

Reply:
> I can fully understand, but here the main point is only one parameter "UEFI" is enough for checking vmware guest boot option. In addition, one more parameter will cause the tests confused, e.g. Other tests like JeOS doesn't give the parameter "UEFI", but for some reason "IPXE_UEFI" is set to 1. If it's really necessary checking more parameters, then we can create a separate function for instead.

This is not the case. is_uefi_boot hide the details away, it detects either parameter to be used, so you just call it and its name tells what to be judged. There is no confusion actually. You can keep using the parameter picked by you and is_uefi_boot does the work for you. It is better and convenient.

Reply:
@alice-suse, Do we have plan to support ipxe boot for vmware guest VM in future? If yes, I will do it with "IPXE_UEFI". If not, I think it's not the responsibility of this PR, and it's meaningless for current vmware tests. I'd like to keep the code line simple and easy.

Reply:
@nanzhg If I understand correctly, current vmware guest installation test relies on ISO. It is easier than pxe installation, so I do not see need to switch to ipxe.

Reply:
This has nothing to do with whether vmware supports ipxe. It is about using a function is_uefi_boot to check whether uefi boot is in place. So is_uefi_boot checks all eligible parameters, including UEFI, IPXE_UEFI. @nanzhg

Reply:
I don't think it requires to use a function for checks in this vmware test, and also I don't see the value of that. So I 'd like to keep the current code.